### ISSUE TITLE
Issue with training of student

### ISSUE THREAD

**Issue by gauravharjule** [Hao840/OFAKD#27]
Hiiie the work is fantastic
Was trying my hands on distillation using your code
Did following till now and need your guidance to move forward pls
1. I have used the pretrained model given at the link your code of deit_tiny_patch16_224 and fine tuned it on CIFAR 100 for around 50 epochs to get an accuracy of 76% approx with following code
import os
import urllib.request
import tarfile
from torchvision import datasets, transforms
from PIL import Image

# Define the URL for CIFAR-100 dataset
cifar100_url = "https://www.cs.toronto.edu/~kriz/cifar-100-python.tar.gz"

# Define the target directory for the CIFAR-100 dataset
target_dir = '/content/datset'

# Create the target directory if it doesn't exist
os.makedirs(target_dir, exist_ok=True)

# Download and extract the CIFAR-100 dataset
tar_file_path = os.path.join(target_dir, 'cifar-100-python.tar.gz')
urllib.request.urlretrieve(cifar100_url, tar_file_path)

# Extract the contents of the tar file
with tarfile.open(tar_file_path, 'r:gz') as tar:
    tar.extractall(target_dir)

# Define the transformations for the images
transform = transforms.Compose([
    transforms.ToTensor(),
    transforms.Normalize((0.5071, 0.4865, 0.4409), (0.2673, 0.2564, 0.2762))  # Normalize the images
])

# Load the CIFAR-100 training dataset
train_dataset = datasets.CIFAR100(root=target_dir, train=True, download=False, transform=transform)

# Save preprocessed images as JPG files for visualization
save_dir_train = os.path.join(target_dir, 'preprocessed_images', 'train')
os.makedirs(save_dir_train, exist_ok=True)

for i, (image, label) in enumerate(train_dataset):
    image = transforms.ToPILImage()(image)
    image.save(os.path.join(save_dir_train, f'image_{i}_label_{label}.jpg'))

# Load the CIFAR-100 testing dataset
test_dataset = datasets.CIFAR100(root=target_dir, train=False, download=False, transform=transform)

# Save preprocessed images as JPG files for visualization
save_dir_test = os.path.join(target_dir, 'preprocessed_images', 'test')
os.makedirs(save_dir_test, exist_ok=True)

for i, (image, label) in enumerate(test_dataset):
    image = transforms.ToPILImage()(image)
    image.save(os.path.join(save_dir_test, f'image_{i}_label_{label}.jpg'))

2. Now I tried to do the distillation using the following wherein I used the finetuned model generated from above code-- !python -m torch.distributed.launch --nproc_per_node=1 train.py /content/datset/preprocessed_images/ --config configs/cifar/cnn.yaml --model resnet18 --teacher deit_tiny_patch16_224 --teacher-pretrained /content/fine_tuned_deit_model.pth --distiller ofa --ofa-eps 1.5
3. now the issue is the test accuracy from 1st epoch I am getting is 0% thats strange and stops me from  further progress
4. I have attached the relevant screenshots just for your ref
5. hoping to hear from you soon
6. 
![image](https://github.com/Hao840/OFAKD/assets/129313849/97c6fcaa-deaa-4067-88ce-d9028ae23a5d)

**Comment by gauravharjule** [comment]
So far in the code which I have seen only one checkpt is given that too for the deit_tiny_patch16_224
are there any other check pts you have 

request pls share them

**Comment by Hao840** [comment]
I think maybe there are something wrong with your `num_classes` setting.

**Comment by gauravharjule** [comment]
I have already changed it to 100 in num of classes

**Comment by Becai** [comment]
Pls where is num_classes modified

**Comment by gauravharjule** [comment]
you can change it in train.py
*gauravharjule*

**Comment by Becai** [comment]
Is it to use the args "--num-classes" in train.py

**Comment by gauravharjule** [comment]
Yes, it is.
*gauravharjule*

**Comment by Becai** [comment]
Thank you for your patient reply, of course we can.

**Comment by gauravharjule** [comment]
how can we connect?

**Comment by Becai** [comment]
yxin7even@gmail.com
